spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/security/CognitoAuthenticationProperties.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.autoconfigure.security;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Cognito Authentication properties.
+ *
+ * @author Eddú Meléndez
+ * @since 2.3
+ */
+@ConfigurationProperties(prefix = "spring.cloud.aws.security.cognito")
+public class CognitoAuthenticationProperties {
```

Comment:
nullability annotations are missing. Each package should have `package-info.java` marking everything as `@NonNull` by default, and all nullable fields, method parameters and return values should be marked with `@Nullable`.

Reply:
TBH right now I don't see value adding those annotations in configuration properties. I agree if those are added in public apis such as *template

Reply:
If we don't add them, everything is `@NonNull` by default. While not such a big deal for Java, when used with Kotlin non-null fields are seen as non-nullable types which can cause very unexpected NPEs in Kotlin Land